src/core/main.c

```diff
@@ -1952,6 +1959,9 @@ static int load_configuration(int argc, char **argv, const char **ret_error_mess
                 return r;
         }
 
+        /* Log level should be put in effect as soon as possible. */
+        log_set_max_level(arg_default_log_max_level);
```

Comment:
hmm, thinking about this, this is not really OK to do this way, as this way you disable the effect of  log_parse_environment(), and override the log level unconditionally back to LOG_INFO, even if it was raised through an env var...

Reply:
hmm, I think arg_default_log_max_level and arg_default_log_target should default to -1/_LOG_TARGET_INVALID by default, so that we can easily check if they were explicitly set by user configuration, and only then put them into effect.

Also, I figure we should propagate into the manager object the log level that is actually set at that time instead of whatever is in arg_default_log_max_level(). i.e. manager_new() should just initialize the target and level field from log_get_max_level() and log_get_target() if you follow what i mean?